src/05-derivatives/03-imaging.md

```diff
@@ -0,0 +1,368 @@
+| Atlas        | OPTIONAL. Which atlas (if any) was used to derive the segmentation.                                                                    |
+| Resolution   | REQUIRED if `res` is present. String, or [object][] mapping labels to strings. Specifies the interpretation of the resolution keyword. |
+| Density      | REQUIRED if `den` is present. String, or [object][] mapping labels to strings. Specifies the interpretation of the density keyword.    |
+
+### Discrete Segmentations
+
+Discrete segmentations of brain tissue represent each tissue class with a unique
+integer label in a 3D volume. See [Anatomical Labels](#anatomical-labels) for interpretation how
+integer values map to tissue classes.
+
+Template:
+
+```Text
+<pipeline_name>/
+    sub-<participant_label>/
+        anat|func|dwi/
+            <source_entities>[_space-<space>][_res-<label>][_den-<label>]_dseg.nii.gz
+```
+
+Example:
+
+```Text
+pipeline/
+    sub-001/
+        anat/
+            sub-001_space-orig_dseg.nii.gz
+            sub-001_space-orig_dseg.json
+```
+
+A segmentation could be a binary mask that functions as a discrete `label` for a
+single structure. In this case, the label key must be used to specify the
+corresponding structure. For example:
+
+```Text
+pipeline/
+    sub-001/
+        anat/
+            sub-001_space-orig_label-GM_dseg.nii.gz
+```
+
+See [Anatomical Labels](#anatomical-labels) for reserved key values for `label`.
+
+### Probabilistic Segmentations
+
+Probabilistic segmentations of brain tissue represent a single tissue class with
+values ranging from 0 to 1 in individual 3D volumes or across multiple frames.
+Similarly to a discrete, binary segmentation, the label key can be used to
+specify the corresponding structure.
+
+Template:
+
+```Text
+<pipeline_name>/
+    sub-<participant_label>/
+        func|anat|dwi/
+            <source_entities>[_space-<space>][_res-<label>][_den-<label>][_label-<label>]_probseg.nii.gz
+```
+
+Example:
+
+```Text
+pipeline/
+    sub-001/
+        anat/
+            sub-001_space-orig_label-BG_probseg.nii.gz
+            sub-001_space-orig_label-WM_probseg.nii.gz
+```
+
+See [Anatomical labels](#anatomical-labels) for reserved key values for `label`.
+
+A 4D probabilistic segmentation, in which each frame corresponds to a different
+tissue class, must provide a label mapping in its JSON sidecar. For example:
+
+```Text
+pipeline/
+    sub-001/
+	    anat/
+		    sub-001_space-orig_probseg.nii.gz
+		    sub-001_space-orig_probseg.json
+```
+
+The JSON sidecar must include the label-map key that specifies a tissue label
+for each volume:
+
+```JSON
+{
+    "LabelMap": [
+        "BG",
+        "WM",
+        "GM"
+        ]
+}
+```
+
+Values of `label` need to map to Abbreviations defined in [Anatomical Labels](#anatomical-labels).
+
+### Surface Parcellations
```

Comment:
While I've not had the experience with working directly with the available software tools here, I feel like on a conceptual level the hierarchical structure here is wrong.

Currently you have:
- Discrete segmentations
- Probabilistic segmentations
- Surface parcellations

This, for example, forbids the possibility of probabilistic parcellations on a surface.

It seems to me the more appropriate separation of concepts (at least on a fundamental theoretical level) is:

- On what *manifold* is the segmentation / parcellation defined?
    - Voxels
    - Surface
    - Combined
- What is the nature of the *data type* encoded?
    - Integer labels
    - Floating-point between 0 and 1

Reply:
I think the issue here is that GIFTI and CIFTI-2 do not have any accommodation for probabilistic surface parcellations, so there are no recommendations to make at this time. They can obviously be hacked, but BIDS tries to follow and harmonize existing practices, rather than impose them from a theoretical perspective.

That said, I agree with the hierarchy you describe.

Would it address your concerns to have a prefatory paragraph explaining this conceptual hierarchy, and that we cover the (more familiar) volumetric cases first, and then the integer labels on surface/combined manifolds?

Reply:
> do not have any accommodation for probabilistic surface parcellations

GIFTI can store values that are probabilistic surface parcellation (e.g., the BAX files from freesurfer are all probabilistic parcels of brodmann areas). technically CIFTI-2 could store this as well in a dense scalar file.

Reply:
more generally to extend out this section, perhaps @ahoopes can describe how the probabilistic maps from freesurfer were being considered in the derivatives context.

Reply:
@satra If they can accommodate, then I'm happy to extend to `_probseg`, but the specs do not seem very accommodating, and there's no obvious extension to use.

In CIFTI-2, it seems the two-part extension is required, but there is some flexibility:

> the NIfTI intent codes and their corresponding intent names.However, it is permissible to use a combination of mapping types that is not in this list.  When doing so, the NIfTI intent code should be the special code NIFTI_INTENT_CONNECTIVITY_UNKNOWN, which has a value of 3000, it should have an intent_name of “ConnUnknown” in the NIfTI-2 header, and the filename should have an extension of the form “.something.nii” , with “something” replaced by a unique string for the mapping combination, of the user’s choice.  If a standardized intent code and extension are desired for a particular combination of mappings, the CIFTI working group should be contacted (hcp-cifti@humanconnectome.org) so that it can be incorporated into the next revision of CIFTI.

I would propose that we defer the specification of this to structural derivatives and try to actively engage the CIFTI working group.

For GIFTI, do you have examples in practice? I'm unable to `mris_convert` a BA label file to a GIFTI. I think we can hack it with the current spec, as it allows `.gii` with no additional extension component is acceptable, and we can always use `NIFTI_INTENT_NONE`.

Reply:
@effigies - hence the ping to andrew. there is no difference really between a thickness map and a probabilistic map from the gifti format perspective. so technically this is possible, but i don't know how the thought process went for the freesurfer derivatives into some standardized formats. regarding actually encoding that these are probabilistic images internally, that's indeed a different conversation. but bids can encode them externally.

Reply:
Sorry, my GitHub isn't live-updating, so I didn't see your ping.

Reply:
Resolving here. These recommendations can be added without breaking backwards compatibility in BEP 011 (structural derivatives).